components/ws-manager-bridge/src/bridge.ts

```diff
@@ -195,176 +202,185 @@ export class WorkspaceManagerBridge implements Disposable {
             //                                The workspace ID of ws-manager is the workspace instance ID in the database.
             //                                The meta ID of ws-manager is the workspace ID in the database.
             const instanceId = status.id!;
-            const workspaceId = status.metadata!.metaId!;
-            const userId = status.metadata!.owner!;
+            const workspaceId = metadata.metaId;
+            const userId = metadata.owner;
             const logContext: LogContext = {
                 userId,
                 instanceId,
                 workspaceId,
             };
 
-            const instance = await this.workspaceDB.trace({ span }).findInstanceById(instanceId);
-            if (instance) {
-                this.metrics.statusUpdateReceived(this.cluster.name, true);
-            } else {
-                // This scenario happens when the update for a WorkspaceInstance is picked up by a ws-manager-bridge in a different region,
-                // before periodic deleter finished running. This is because all ws-manager-bridge instances receive updates from all WorkspaceClusters.
-                // We ignore this update because we do not have anything to reconcile this update against, but also because we assume it is handled
-                // by another instance of ws-manager-bridge that is in the region where the WorkspaceInstance record was created.
-                this.metrics.statusUpdateReceived(this.cluster.name, false);
-                return;
-            }
-
-            const currentStatusVersion = instance.status.version || 0;
-            if (currentStatusVersion > 0 && currentStatusVersion >= status.statusVersion) {
-                // We've gotten an event which is older than one we've already processed. We shouldn't process the stale one.
-                span.setTag("statusUpdate.staleEvent", true);
-                this.metrics.recordStaleStatusUpdate();
-                log.debug(ctx, "Stale status update received, skipping.");
-            }
-
-            if (!!status.spec.exposedPortsList) {
-                instance.status.exposedPorts = status.spec.exposedPortsList.map((p) => {
-                    return <WorkspaceInstancePort>{
-                        port: p.port,
-                        visibility: mapPortVisibility(p.visibility),
-                        protocol: mapPortProtocol(p.protocol),
-                        url: p.url,
+            await this.redisMutex.lockWorkspaceStatusUpdate(workspaceId, 2 * 1000, async () => {
```

Comment:
Why do we need this mutex here? :thinking:

Reply:
to not overwrite the status concurrently

Reply:
But that seems to be disabled by the feature flag here already, no?
Then we only have the concurrent writers on the server side (which brings us to [this discussion](https://github.com/gitpod-io/gitpod/pull/18323/files#r1277319318)).

Reply:
Both sides are currently locked because the phase as well as the git status is on the same property and would overwrite each other. There is no need for locking on the update git status if we resolve this by introducing another property.

Reply:
Gotcha, missed the phase in the right-shift diff. :smile: